### ISSUE TITLE
SDK does not work with 3.201.1.0 (2024-12-12) update

### ISSUE THREAD

**Issue by crudson** [OrfeasZ/ZHMModSDK#138]
if dinput8 is enabled (as per instructions for linux) then game crashes after options screen (via steam w/ GE-Proton9-20)

**Comment by krezovic** [comment]
on Windows the game won't even start

**Comment by OrfeasZ** [comment]
I'm aware and trying to fix this. Unfortunately the latest update broke basically everything because of what looks like a new optimization pass / different compiler settings. This might take a while...

**Comment by brianwalleshauser** [comment]
To fix the game not loading, simply delete the dinput8.dll from your HITMAN3\Retail directory

**Comment by amaturebutokrn** [comment]
> on Windows the game won't even start

You can get to the settings, but after that the game just crashes (Steam on Windows)

**Comment by amaturebutokrn** [comment]
> I'm aware and trying to fix this. Unfortunately the latest update broke basically everything because of what looks like a new optimization pass / different compiler settings. This might take a while...

Not to be rude, because I know you have stuff other then this to worry about, but do you have a timeframe on when it will be able to be fixed )if it can). this SDK is a pretty decent mod and works really well (no other stuff needs to be downloaded ect.) btw, so thx. Just curious...

**Comment by irfansyakir** [comment]
For now, I am still able to get the SDK working by downpatching to an earlier version of Hitman
https://hitruns-wiki.vercel.app/docs/downpatching
I downpatched to version 3.170.1 (3.180 works too i believe) of Hitman 3 and the SDK is working properly, but release v2.0.0 was buggy for me, I used v1.8.0.

**Comment by ewenlau** [comment]
> For now, I am still able to get the SDK working by downpatching to an earlier version of Hitman https://hitruns-wiki.vercel.app/docs/downpatching I downpatched to version 3.170.1 (3.180 works too i believe) of Hitman 3 and the SDK is working properly, but release v2.0.0 of the SDK was buggy for me, so I am using v1.8.0.

Well, that's the entire issue. It does work with earlier versions, but not the current one.

**Comment by irfansyakir** [comment]
> > For now, I am still able to get the SDK working by downpatching to an earlier version of Hitman https://hitruns-wiki.vercel.app/docs/downpatching I downpatched to version 3.170.1 (3.180 works too i believe) of Hitman 3 and the SDK is working properly, but release v2.0.0 of the SDK was buggy for me, so I am using v1.8.0.
> 
> Well, that's the entire issue. It does work with earlier versions, but not the current one.

yup, just pointing this out if people don't want to wait for a new SDK release and use it right now. i only discovered this SDK right after the update and I couldn't wait for an update, so i went and see if it was possible to downgrade to an earlier version and see if the SDK would work, and it turns out it did

**Comment by Yaqps** [comment]
can we just downgrade zhm? Would that work? I dont want to downgrade hitman for it

**Comment by Yaqps** [comment]
![image](https://github.com/user-attachments/assets/71fe976e-9647-4b5f-9ba3-218b45257c90)
I think I broke hitman

**Comment by gameplayoffert** [comment]
> can we just downgrade zhm? Would that work? I dont want to downgrade hitman for it I was not able to find how to do it

This is the point of downgrading Hitman :D, it's because the latest SDK code doesn't work with the latest hitman version, however it does work on the previous hitman version. If you downgrade the SDK, it won't solve the problem because the code of the SDK will be even older than the latest code and Hitman will stay in it's current state so there is no point of doing that. So about your question "Would that work" then the answer is no. I didn't try to downgrade Hitman, but no matter what, i agree that it's really annoying, the SDK is really an awesome thing.

**Comment by Thomas139865** [comment]
> To fix the game not loading, simply delete the dinput8.dll from your HITMAN3\Retail directory

yes but then the mod doesnt work

**Comment by amaturebutokrn** [comment]
> > > To fix the game not loading, simply delete the dinput8.dll from your HITMAN3\Retail directory
> > 
> > 
> > yes but then the mod doesnt work
> 
> Some mods will still function even though the Mod SDK does not currently. So you can temporarily rename the file so the game can run, you can add either `.old` or `.bkp` on the end of the DLL filename until the Mod SDK gets a new release which is probably gonna take some time before then.

so does that fix the issue temporarily, or just make it so that you dont have to delete the .dll?

**Comment by GauthierPLM** [comment]
I built a release from source that works on my side:
[db5532d31e84eef54fc7fcf93df996c85d64238c-release.zip](https://github.com/user-attachments/files/18246879/db5532d31e84eef54fc7fcf93df996c85d64238c-release.zip)

⚠️ This is not an official build! Don't expect any support! Things may not work as expected.
⚠️ If you have any doubt, wait for **an official release** or build it yourself by following the wiki' instructions.

**Comment by OrfeasZ** [comment]
Please don't share random binaries. If you absolutely need a development / pre-release version of the SDK and its mods you can grab it from the [automatic builds that happen after every commit](https://github.com/OrfeasZ/ZHMModSDK/actions/runs/12497804105), though I'd recommend you wait until a proper release.

**Comment by OrfeasZ** [comment]
A new version of the SDK is now released with support for the latest game update. You can download it [from here](https://github.com/OrfeasZ/ZHMModSDK/releases/tag/v3.0.0). Keep in mind that any third-party SDK mods you use (eg. Freelancer Variations) will need to be updated too before you can use them.
